l3kernel/l3keys.dtx

```diff
@@ -1822,14 +1822,14 @@
 %   }
 %   Searching for a property means finding the last |.| in the input,
 %   and storing the text before and after it. Everything is turned into
-%   strings, so there is no problem using an \texttt{e}-type expansion. Since
-%   \cs{@@_trim_spaces:n} will turn its argument into a string anyway, this
-%   function uses \cs{cs_set_nopar:Npe} instead of \cs{tl_set:Ne} to gain some
-%   speed.
+%   strings at first, so there is no problem using an \texttt{e}-type expansion
+%   in aux functions. \cs{cs_set_nopar:Npe} instead of \cs{tl_set:Ne} or
+%   \cs{str_set:Ne} is used to gain some speed.
+%   Space trimming on key path is done only once, after the last |.| is found.
 %    \begin{macrocode}
 \cs_new_protected:Npn \@@_property_find:n #1
   {
-    \cs_set_nopar:Npe \l_@@_property_str { \@@_trim_spaces:n { #1 } }
+    \str_set:Nn \l_@@_property_str { #1 }
```

Comment:
You've dropped the `\cs_set_nopar:Npe` here, so the text comment doesn't make sense - and do we not still want the extra speed?

Reply:
The comment on `\cs_set_nopar:Npe` is actually for aux functions. I can make this clearer if needed.

After #1349, [`\str_set:Nn`][new str_set:Nn] is defined by
```
\cs_new_protected:Npn \str_set:Nn #1#2
  { \__kernel_tl_set:Ne #1 { \__kernel_tl_to_str:w {#2} } }
```
hence
```
\str_set:Nn \l_@@_property_str { #1 }
```
is only one expansion step slower than
```
\cs_set_nopar:Npe \l_@@_property_str { \tl_to_str:n { #1 } }
```

[new str_set:Nn]: https://github.com/latex3/latex3/blob/0ec3512fc5a80ba97752d1d852d6782fe140dbb3/l3kernel/l3str.dtx#L1001-L1002